### ISSUE TITLE
eventd-0.24.0 fails test - ERROR:../eventd-0.24.0/src/libnkutils/tests/token.c:972:_nk_token_list_tests_func: assertion failed (result == data->result): ("1,00000" == "1.00000")

### ISSUE THREAD

**Issue by Coacher** [sardemff7/eventd#39]
Hello.

eventd-0.24.0 fails test:
```
>>> Test phase: net-misc/eventd-0.24.0-r9999
ninja -v -j4 -l4 -C /var/tmp/portage/net-misc/eventd-0.24.0-r9999/work/eventd-0.24.0-build test
ninja: Entering directory `/var/tmp/portage/net-misc/eventd-0.24.0-r9999/work/eventd-0.24.0-build'
[1/33] /var/tmp/portage/net-misc/eventd-0.24.0-r9999/work/eventd-0.24.0-build/src/libnkutils/nk-git-version header src/libnkutils/nkutils-git-version.h /var/tmp/portage/net-misc/eventd-0.24.0-r9999/work/eventd-0.24.0 /var/tmp/portage/net-misc/eventd-0.24.0-r9999/work/eventd-0.24.0/.git /usr/bin/git
[2/12] /var/tmp/portage/net-misc/eventd-0.24.0-r9999/work/eventd-0.24.0-build/src/libnkutils/nk-git-version entity src/libnkutils/nkutils-git-version.ent /var/tmp/portage/net-misc/eventd-0.24.0-r9999/work/eventd-0.24.0 /var/tmp/portage/net-misc/eventd-0.24.0-r9999/work/eventd-0.24.0/.git /usr/bin/git
[2/3] /usr/bin/python3.5 -u /usr/lib/python-exec/python3.5/meson test --no-rebuild --print-errorlogs
 1/14 libnkutils / libnkutils uuid module tests  OK       0.05 s
 2/14 libnkutils / libnkutils token module tests  FAIL     0.10 s
 3/14 libnkutils / libnkutils gtk-settings module tests  OK       0.06 s
 4/14 libnkutils / libnkutils xdg-de module tests  OK       0.33 s
 5/14 libnkutils / libnkutils xdg-theme module tests  OK       0.21 s
 6/14 libnkutils / libnkutils enum module tests  OK       0.04 s
 7/14 libnkutils / libnkutils colour module tests  OK       0.04 s
 8/14 libnkutils / libnkutils bindings module tests  OK       1.12 s
 9/14 eventd / libeventd-event unit tests     OK       0.23 s
10/14 eventd / libeventd-protocol unit tests  OK       0.03 s
11/14 eventd / eventd EvP integration test    OK       0.40 s
12/14 eventd / eventd relay integration test  OK       0.52 s
13/14 eventd / eventd ws integration test     OK       0.49 s
14/14 eventd / libeventc integration test     OK       0.28 s

OK:        13
FAIL:       1
SKIP:       0
TIMEOUT:    0


The output from the failed tests:

 2/14 libnkutils / libnkutils token module tests  FAIL     0.10 s

--- command ---
/var/tmp/portage/net-misc/eventd-0.24.0-r9999/work/eventd-0.24.0-build/src/libnkutils/nk-token.test
--- stdout ---
/nkutils/token/basic: OK
/nkutils/token/basic/dash-underscore: OK
/nkutils/token/basic/unicode: OK
/nkutils/token/basic/boolean/true: OK
/nkutils/token/basic/boolean/false: OK
/nkutils/token/basic/wrong/1: OK
/nkutils/token/basic/wrong/2: OK
/nkutils/token/key/name: OK
/nkutils/token/key/index/positive: OK
/nkutils/token/key/index/negative: OK
/nkutils/token/key/name/modifier: OK
/nkutils/token/key/join/default: OK
/nkutils/token/key/join/custom: OK
/nkutils/token/key/join/replace: OK
/nkutils/token/wrong/modifier: OK
/nkutils/token/wrong/key: OK
/nkutils/token/wrong/key/index: OK
/nkutils/token/wrong/regex/pattern: OK
/nkutils/token/wrong/regex/replace: OK
/nkutils/token/fallback/with: OK
/nkutils/token/fallback/without: OK
/nkutils/token/fallback/recurse: OK
/nkutils/token/fallback/boolean/true: OK
/nkutils/token/fallback/boolean/false: OK
/nkutils/token/substitute/with: OK
/nkutils/token/substitute/without: OK
/nkutils/token/substitute/boolean/true: OK
/nkutils/token/substitute/boolean/false: OK
/nkutils/token/anti-substitute/with: OK
/nkutils/token/anti-substitute/without: OK
/nkutils/token/anti-substitute/boolean/true: OK
/nkutils/token/anti-substitute/boolean/false: OK
/nkutils/token/range/symbol: OK
/nkutils/token/range/text: OK
/nkutils/token/range/middle-split: OK
/nkutils/token/range/double: OK
/nkutils/token/range/plural/singular: OK
/nkutils/token/range/plural/plural: OK
/nkutils/token/prettify/float/width: OK
/nkutils/token/prettify/float/0-padding: OK
/nkutils/token/prettify/float/precision: FAIL
/nkutils/token/prettify/prefixes/binary: FAIL
/nkutils/token/prettify/prefixes/si/big: OK
/nkutils/token/prettify/prefixes/si/small: OK
/nkutils/token/prettify/prefixes/si/with-precision: FAIL
/nkutils/token/prettify/prefixes/si/2-precision: FAIL
/nkutils/token/prettify/prefixes/si/0-precision: OK
/nkutils/token/prettify/time/default: OK
/nkutils/token/prettify/time/with-format: OK
/nkutils/token/prettify/duration/default: OK
/nkutils/token/prettify/duration/with-format: OK
/nkutils/token/replace/full: OK
/nkutils/token/replace/missing: OK
/nkutils/token/replace/capture: OK
/nkutils/token/replace/remove: OK
/nkutils/token/replace/multiple: OK
/nkutils/token/replace/before-after/with: OK
/nkutils/token/replace/before-after/without: OK
/nkutils/token/replace/braces/paired: OK
/nkutils/token/replace/braces/opening: OK
/nkutils/token/replace/braces/closing: OK
/nkutils/token/replace/escaping/backslash: OK
/nkutils/token/replace/escaping/forwardslash/1: OK
/nkutils/token/replace/escaping/forwardslash/2: OK
/nkutils/token/replace/escaping/forwardslash/3: OK
/nkutils/token/replace/escaping/right-curly-bracket/2: OK
/nkutils/token/replace/recurse/with: OK
/nkutils/token/old/before-after: OK
/nkutils/token/identifier/double-escape: OK
/nkutils/token/identifier/non-dollar: OK
/nkutils/token/identifier/none: OK
/nkutils/token/enum/basic: OK
/nkutils/token/enum/nested: OK
/nkutils/token/enum/fallback: OK
/nkutils/token/enum/substitute: OK
/nkutils/token/enum/anti-substitute: OK
/nkutils/token/enum/regex: OK
/nkutils/token/enum/prettify/float: FAIL
/nkutils/token/enum/prettify/prefixes/si: OK
/nkutils/token/enum/prettify/prefixes/binary: OK
/nkutils/token/enum/wrong/regex: OK
/nkutils/token/enum/wrong/token: OK
--- stderr ---
**
ERROR:../eventd-0.24.0/src/libnkutils/tests/token.c:972:_nk_token_list_tests_func: assertion failed (result == data->result): ("1,00000" == "1.00000")
**
ERROR:../eventd-0.24.0/src/libnkutils/tests/token.c:972:_nk_token_list_tests_func: assertion failed (result == data->result): ("612,015625Ki" == "612.015625Ki")
**
ERROR:../eventd-0.24.0/src/libnkutils/tests/token.c:972:_nk_token_list_tests_func: assertion failed (result == data->result): ("1,0M" == "1.0M")
**
ERROR:../eventd-0.24.0/src/libnkutils/tests/token.c:972:_nk_token_list_tests_func: assertion failed (result == data->result): ("626,70k" == "626.70k")
**
ERROR:../eventd-0.24.0/src/libnkutils/tests/token.c:1166:_nk_token_list_enum_tests_func: assertion failed (result == data->result): ("1,56" == "1.56")
-------

Full log written to /var/tmp/portage/net-misc/eventd-0.24.0-r9999/work/eventd-0.24.0-build/meson-logs/testlog.txt
FAILED: meson-test
/usr/bin/python3.5 -u /usr/lib/python-exec/python3.5/meson test --no-rebuild --print-errorlogs
ninja: build stopped: subcommand failed.
```

Full test log: https://paste.pound-python.org/show/VrdzkeB1cOEmTnZ2cbOd/
Full build log: https://paste.pound-python.org/show/JGjDQ7mLipS4fdm6mxpj/

**Comment by sardemff7** [comment]
Fixed in sardemff7/libnkutils@04362cb6860715d2dd258787168e35bf3b54c714
In the mean time, please export `LC_NUMERIC=C` for eventd tests.

**Comment by Coacher** [comment]
Thank you for the workaround.

**Comment by Coacher** [comment]
Fixed in 0.24.1. Thank you.
